core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -0,0 +1,197 @@
+                })?;
+        let validated = transaction
+            .prepare_and_validate(&TransactionValidator::new_with_latest_config(
+                &state.network,
+            ))
+            .map_err(|e| {
+                ResponseError::from(ApiError::InvalidTransaction)
+                    .with_details(format!("Invalid transaction: error = {:?}", e))
+            })?;
+
+        let instructions = transaction.signed_intent.intent.instructions.0;
+        let signers = validated.signer_keys;
+        (instructions, signers)
+    } else {
+        let prepared_intent = PreparedIntentV1::prepare(
+            &RawTransactionIntent::from_vec(transaction_bytes),
+            &PreparationSettings::latest(),
+        )
+        .unwrap();
+        #[allow(deprecated)]
+        let instructions = prepared_intent.instructions.inner.0;
+        let signers = index_set_new();
+        (instructions, signers)
+    };
+
+    let database = state.state_manager.database.snapshot();
+    let operations = parse_instructions(
+        &instructions,
+        &MappingContext::new(&state.network),
+        database.deref(),
+    )?;
+
+    // See https://docs.cdp.coinbase.com/mesh/docs/models#constructionparseresponse for field
+    // definitions
+    Ok(Json(models::ConstructionParseResponse {
+        operations,
+        signers: None,
+        account_identifier_signers: Some(
+            signers
+                .into_iter()
+                .map(|x| AccountIdentifier {
+                    address: state.public_key_to_address_string(x),
+                    sub_account: None,
+                    metadata: None,
+                })
+                .collect(),
+        ),
+        metadata: None,
+    }))
+}
+
+pub fn parse_instructions(
+    instructions: &[InstructionV1],
+    mapping_context: &MappingContext,
+    database: &StateManagerDatabase<impl ReadableRocks>,
+) -> Result<Vec<Operation>, ResponseError> {
+    let mut operations = Vec::new();
+    let mut next_index = 0;
+    while next_index < instructions.len() {
+        let mut instruction = &instructions[next_index];
+        next_index = next_index + 1;
+        match instruction {
+            InstructionV1::CallMethod(CallMethod {
+                address: DynamicGlobalAddress::Static(global_address),
+                method_name,
+                args,
+            }) => {
+                let args_bytes = manifest_encode(&args).unwrap();
+                match method_name.as_str() {
+                    "lock_fee" => (),
+                    "withdraw" => {
+                        let input = manifest_decode::<AccountWithdrawManifestInput>(&args_bytes)
+                            .map_err(|_| {
+                                ResponseError::from(ApiError::InvalidWithdrawInstruction)
+                                    .with_details("Invalid withdraw instruction")
+                            })?;
+                        operations.push(Operation {
```

Comment:
We could probably have a `to_api_operation(context: &MappingContenxt, operation_index: usize, operation_type: MeshApiOperationType, global_address: GlobalAddress, amount: Decimal, status: Option<MeshApiOperationStatus>)` conversion in `conversions/operations.rs`

Reply:
Done